src/careamics/transforms/pixel_manipulation_torch.py

```diff
@@ -0,0 +1,406 @@
+        rng = torch.default_generator
+
+    # Relative axis
+    moving_axis = -1 - struct_params.axis
+
+    # Create a mask array
+    mask_shape = [1] * len(patch.shape)
+    mask_shape[moving_axis] = struct_params.span
+    mask = torch.ones(mask_shape, device=patch.device)
+
+    center = torch.tensor(mask.shape, device=patch.device) // 2
+
+    # Mark the center
+    mask[tuple(center)] = 0
+
+    # Displacements from center
+    displacements = torch.stack(torch.where(mask == 1)) - center.unsqueeze(1)
+
+    # Combine all coords (ndim, npts) with all displacements (ncoords, ndim)
+    mix = displacements.T.unsqueeze(-1) + coords.T.unsqueeze(0)
+    mix = mix.permute([1, 0, 2]).reshape([mask.ndim, -1]).T
+
+    # Filter out invalid indices
+    valid_indices = (mix[:, moving_axis] >= 0) & (
+        mix[:, moving_axis] < patch.shape[moving_axis]
+    )
+    mix = mix[valid_indices]
+
+    # Replace neighboring pixels with random values from a uniform distribution
+    random_values = torch.empty(len(mix), device=patch.device).uniform_(
+        patch.min().item(), patch.max().item(), generator=rng
+    )
+    patch[tuple(mix.T.tolist())] = random_values
+
+    return patch
+
+
+def _odd_jitter_func_torch(step: float, rng: torch.Generator) -> torch.Tensor:
+    """
+    Randomly sample a jitter to be applied to the masking grid.
+
+    This is done to account for cases where the step size is not an integer.
+
+    Parameters
+    ----------
+    step : float
+        Step size of the grid, output of np.linspace.
+    rng : torch.Generator
+        Random number generator.
+
+    Returns
+    -------
+    torch.Tensor
+        Array of random jitter to be added to the grid.
+    """
+    step_floor = torch.floor(torch.tensor(step))
+    odd_jitter = (
+        step_floor
+        if step_floor == step
+        else torch.randint(high=2, size=(1,), generator=rng)
+    )
+    return step_floor if odd_jitter == 0 else step_floor + 1
+
+
+def _get_stratified_coords_torch(
+    mask_pixel_perc: float,
+    shape: tuple[int, ...],
+    rng: Optional[torch.Generator] = None,
+) -> torch.Tensor:
+    """
+    Generate coordinates of the pixels to mask.
+
+    Randomly selects the coordinates of the pixels to mask in a stratified way, i.e.
+    the distance between masked pixels is approximately the same.
+
+    Parameters
+    ----------
+    mask_pixel_perc : float
+        Actual (quasi) percentage of masked pixels across the whole image. Used in
+        calculating the distance between masked pixels across each axis.
+    shape : tuple[int, ...]
+        Shape of the input patch.
+    rng : torch.Generator or None
+        Random number generator.
+
+    Returns
+    -------
+    np.ndarray
+        Array of coordinates of the masked pixels.
+    """
+    if rng is None:
+        rng = torch.default_generator
+
+    # TODO explain this
+    mask_pixel_distance = round((100 / mask_pixel_perc) ** (1 / len(shape)))
+
+    pixel_coords = []
```

Comment:
Any chance you could add comments here to explain the various steps for future us?